### ISSUE TITLE
Wizard Drawing

### ISSUE THREAD

**Issue by tripplyons** [CaseyManning/resource-mmorpg-#1]
The wizard isn't drawing right.

**Comment by Skyyconnect** [comment]
Tripp, let me know if this makes sense, it is what i was trying to explain
at the coder school on Friday:



<------------------------------------------------------->
method:
*public void resources(){ *

starting add resource number must be >0 for the for-loop:

* int addResources = 1;*

* System.out.println("<-------start ----------------->");*

creation of the ArrayList to store resources:

* ArrayList<Integer> Resources = new ArrayList<Integer>();*
* for( int i  =0; i < addResources; i++){*


for-loop adds a resource as long as it is less than total amount:
* System.out.println("<-------For-Loop ----------------->");*
* System.out.println("Resources:"+Resources);*
* System.out.println("addResources:"+addResources);*
* System.out.println("interation number:"+i);*

if resources is < the number added:

* if( Resources.size() < addResources ){*
* System.out.println("<-------IF ----------------->");*
* System.out.println("Resources:"+Resources);*
* System.out.println("addResources:"+addResources);*
* System.out.println("interation number:"+i);*

Do-While checks the iteration then adds 1 and breaks ( the while-condition
keeps the resources true so you cant duplicate or keep making resources):
** *might be a good idea to work in your paint function in here somehow***
* do{*
* Resources.add(addResources);*
* }while(addResources < Resources.lastIndexOf(Resources)&& addResources
<4);*
* System.out.println("<-------Do-While ----------------->");*
* System.out.println("Resources:"+Resources);*
* System.out.println("addResources:"+addResources);*
* System.out.println("interation number:"+i);*

Adds a resource to the resource number:
* addResources++; *
* System.out.println("new addResource number:"+addResources); *

Checks the number to see if < what that resource can produce ( in this case
only 4) and breaks once the condition is met.
* if ( i >1){*
* break; *
* }*
* }*
* }*
* }*
* public static void main(String[] args) {*
* tripp trys = new tripp();*
* trys.resources();*
* }*

*}*






You are going to need to add the resources painting in here somewhere, but
i think it depends on how you have it setup so i will leave you to do that
by yourself. This is the base logic of what we are looking for. If you have
any question just ask.

**Comment by Skyyconnect** [comment]
you can also remove a resource  it before the break statement.

**Comment by Skyyconnect** [comment]
output is attached.

**Comment by Skyyconnect** [comment]
This is your new game engine.

the board (in this case) is the "sprite layer"

Have fun , and import your tile map as your background.


see attached "game" file

Best,

Skyy

**Comment by Skyyconnect** [comment]
actually use this one, It's is better edited. i am going to make a github,
please check there from now on to see changes of this code. I will be
adding to this just the bare. check it here:

https://github.com/Skyyconnect/BlankGameRPG

**Comment by CaseyManning** [comment]
So is there a new repository we are writing the game in?

**Comment by Skyyconnect** [comment]
No, I'm just gonna update the "blank" and and you can check to see any
changes for ideas only . Make your own repository and start from there.
I'll let you know if you should look at the blank for reference. You need
to add your tile map and resource logic to your repository version . My
repository is just for reference. Hope this all helps, I think we will be
really better off with this version of  build, next session we can add
threading and state machine and we should be back to where we were. Just
try to get the game ready for next time please. Be creative and have fun!

**Comment by skyylersiejko** [comment]
tripp,

add this to engine :

<------------------------------------------------------------------->
 private StateMachine state;

public Engine  {
 new Thread(new Runnable()
    {
      public void run()
      {
        for (int i = 0; i < 2; i++)
        {
            //adds to the statemachne
                            state.push(state[0]);

```
                       //wait one second and remove state g
      try{Thread.sleep(1000);  state.pop(state[0]); } // 1 second pause
      catch(Exception e){}
    }
  }
}).start();
```

<-------------------------------------------------->
just add the states in the engine and use state machine to do all the work.
Keep it simple!
